src/blocks.zig

```diff
@@ -371,35 +388,59 @@ pub const Block = packed struct { // MARK: Block
 	}
 };
 
-pub const TouchFunction = fn(block: Block, entity: Entity, posX: i32, posY: i32, posZ: i32, isEntityInside: bool) void;
+// MARK: Tick
+pub var tickFunctions: utils.NamedCallbacks(TickFunctions, TickFunction) = undefined;
+pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void;
+pub const TickFunctions = struct {
+	pub fn replaceWithCobble(_: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void {
+		std.log.debug("Replace with cobblestone at ({d},{d},{d})", .{x, y, z});
+		const cobblestone = parseBlock("cubyz:cobblestone");
+
+		_chunk.updateBlockAndSetChanged(x, y, z, cobblestone);
+
+		const wx = _chunk.super.pos.wx + x;
+		const wy = _chunk.super.pos.wy + y;
+		const wz = _chunk.super.pos.wz + z;
+
+		const userList = main.server.getUserListAndIncreaseRefCount(main.stackAllocator);
+		defer main.server.freeUserListAndDecreaseRefCount(main.stackAllocator, userList);
+		const newBlock: main.renderer.mesh_storage.BlockUpdate = .init(.{wx, wy, wz}, cobblestone);
+		const blockUpdates: [1]main.renderer.mesh_storage.BlockUpdate = .{newBlock};
+		for(userList) |user| {
+			main.network.Protocols.blockUpdate.send(user.conn, blockUpdates[0..]);
+		}
```

Comment:
There is also another problem here:
This does not consider neighboring blocks.
E.g. if you have a torch that is connected to the current block and it gets replaced with air (e.g. leaf decay), then the torch wouldn't be removed.

A better solution would be to use `world.cmpxchgBlock` which handles this as well as the networking code.
You also wouldn't need to lock the chunk mutex anymore (→no chance for deadlocks in the future), since the function will compare the block first and only exchange it when it hasn't changed, in case it has changed you can just ignore the tick event and return.